src/components/tiles/text/text-toolbar.tsx

```diff
@@ -24,74 +20,121 @@ import SuperscriptToolIcon from "../../../assets/icons/text/superscript-text-ico
 import SubscriptToolIcon from "../../../assets/icons/text/subscript-text-icon.svg";
 import NumberedListToolIcon from "../../../assets/icons/text/numbered-list-text-icon.svg";
 import BulletedListToolIcon from "../../../assets/icons/text/bulleted-list-text-icon.svg";
-import InsertVariableCardIcon from "../../../plugins/shared-variables/assets/insert-variable-chip-icon.svg";
 
 
 import "./text-toolbar.sass";
 
-interface IButtonDef {
-  iconName: string;  // icon name for this button.
-  Icon: React.FunctionComponent<React.SVGProps<SVGSVGElement>>; // icon for the button
-  toolTip: string;   // Text for the button's tool-tip.
-  buttonEnabled?: (args: any) => boolean; // Decides when the button should be enabled.
-}
-
 interface IProps extends IFloatingToolbarProps, IRegisterTileApiProps {
-  selectedButtons: string[];
   editor?: Editor;
   textContent?: TextContentModelType,
+  // TODO: Need a type for pluginInstances
+  pluginInstances: Record<string, any>,
+  valueRevision: number;
+}
+
+interface IButtonDef {
+  pluginName?: string,
+  component: ButtonDefComponent
 }
 
 const kShortcutPrefix = isMac() ? "Cmd-" : "Ctrl-";
 
-const buttonDefs: IButtonDef[] = [
-  { iconName: "bold",        Icon: BoldToolIcon,          toolTip: `Bold (${kShortcutPrefix}b)`},
-  { iconName: "italic",      Icon: ItalicToolIcon,        toolTip: `Italic (${kShortcutPrefix}i)`},
-  { iconName: "underline",   Icon: UnderlineToolIcon,     toolTip: `Underline (${kShortcutPrefix}u)`},
-  { iconName: "subscript",   Icon: SubscriptToolIcon,     toolTip: `Subscript`},
-  { iconName: "superscript", Icon: SuperscriptToolIcon,   toolTip: `Superscript`},
-  { iconName: "list-ol",     Icon: NumberedListToolIcon,  toolTip: `Numbered List`},
-  { iconName: "list-ul",     Icon: BulletedListToolIcon,  toolTip: `Bulleted List`},
-];
+// These are the built in tool definitions
+const buttonDefs = new Map<string, IButtonDef>([
+  [ "bold", {
+    component: props => <BuiltInToolbarButton
+      iconName="bold"
+      Icon={BoldToolIcon}
+      toolTip={`Bold (${kShortcutPrefix}b)`}
+      slateType={EFormat.bold}
+      command={() => toggleMark(props.editor, EFormat.bold) }
+      {...props}
```

Comment:
I'm planning to refactor this, so it is closer to the declarative style from before. Initially this explicit component approach was pretty concise and seemed more clear to understand but now it seems too messy.